### PR TITLE
add delete button to events

### PR DESCRIPTION
**Pull request by SamamthaH**
## Description

Describe your changes in detail
added delete functionality for events including a button with a trash icon, a popup confirmation asking if they really want to delete, notification that the event has been successfully deleted, and refreshing the page afterward so the event disappears from the list.

## Types of changes

What types of changes does your code introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (only improve the code quality no change in functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] Issue has been created
- [x] Every file touched is linted
- [ ] Every file touched has tests that cover all the funcitonality with green coverage
- [ ] Documentation is updated (if applicable).

## Issue

Please link the issue here.
https://github.com/COSC447-Directed-Studies/canvas-gamification-ui/issues/60

## Tests

Please describe in detail what tests you added or changed.

## Screenshots (if appropriate):
image
image
image
image

### CODE REVIEW

File: src/app/course/event/event-row/event-row.component.ts [hunk 1]
```diff
@@ -33,4 +43,37 @@ export class EventRowComponent implements OnInit {
...
+     * Delete an event from the course-list.
+     */
+    async deleteEvent(eventId: number) {
+        this.courseEventService.deleteCourseEvent(eventId).subscribe()
+
+        this.notificationsService
+            .show('The event has been deleted successfully.', {
+                status: TuiNotification.Success
+            }).subscribe()
+        this.router.navigateByUrl('/RefreshComponent', {skipLocationChange: true}).then(() => {
```
Comment (keyvankhademi) [inline src/app/course/event/event-row/event-row.component.ts]: what is this `RefreshComponent`? and why do we have 2 navigations in side each other?

File: src/app/course/event/event-row/event-row.component.ts [hunk 2]
```diff
@@ -33,4 +43,37 @@ export class EventRowComponent implements OnInit {
...
+    openDeleteEventDialog(
+        content: PolymorpheusContent<TuiDialogContext>,
+        eventId: number
+    ): void {
+        this.dialogService.open(content, {
+            closeable: false,
+            label: 'Delete Event?'
+        }).subscribe({
+            next: () => this.deleteEvent(eventId)
+        })
```
Comment (keyvankhademi) [inline src/app/course/event/event-row/event-row.component.ts]:
```suggestion
        }).subscribe(() => this.deleteEvent(eventId))
```

**Review by keyvankhademi — CHANGES_REQUESTED** [review]

File: src/app/course/event/event-row/event-row.component.ts (hunk 1 above)
Comment (SamamthaH) [reply src/app/course/event/event-row/event-row.component.ts]: I was trying to refresh the page after an event was deleted and this stack overflow was what I pulled from while trying to at the top comment
https://stackoverflow.com/questions/47813927/how-to-refresh-a-component-in-angular
After I got it working I tried it just with navigating to the same page but it didn't seem to do anything when i tested it

Comment (SamamthaH) [reply src/app/course/event/event-row/event-row.component.ts]: window.location.reload() doesn't work, it pulls up the skeleton again but the deleted event still appears after. location.reload has some funky use cases where refreshing the page before you delete sometimes causes it to fail, I've taken a video of it. I believe RefreshComponent doesn't actually exist, I can't find it in any documentation and when I swapped it out for course/1000 in testing it still worked. I imagine what this is doing is saying navigate away from this page but don't show it and then immediately come back forcing the page to effectively do a total reset.

File: src/app/course/event/event-row/event-row.component.html [hunk 3]
```diff
@@ -118,6 +118,21 @@
                 </button>
             </div>
         </ng-template>
+        <button *ngIf="event.has_edit_permission"
+                (click)="openDeleteEventDialog(deleteDialog, event.id)"
+                [tuiHint]="deleteTooltip"
+                class="tui-space_right-2 tui-space_vertical-2"
```
Comment (keyvankhademi) [inline src/app/course/event/event-row/event-row.component.html]: use tailwind classes

File: src/app/course/event/event-row/event-row.component.html [hunk 4]
```diff
@@ -152,3 +167,26 @@
...
+<ng-template #deleteDialog let-observer>
+    <p>Are you sure that you would like to delete this assessment?</p>
+    <div class="tui-form__buttons">
+        <button
+            (click)="observer.next(); observer.complete()"
+            class="tui-form__button"
+            size="m"
+            tuiButton
+        >
+            Confirm Changes
```
Comment (keyvankhademi) [inline src/app/course/event/event-row/event-row.component.html]: Maybe `Confirm Delete`?

File: src/app/course/event/event-row/event-row.component.ts (hunk 1 above)
Comment (keyvankhademi) [reply src/app/course/event/event-row/event-row.component.ts]: We need a better solution, let's discuss it during the meeting.

**Review by keyvankhademi — CHANGES_REQUESTED** [review]

File: src/app/course/event/event-row/event-row.component.html (hunk 3 above)
Comment (SamamthaH) [reply src/app/course/event/event-row/event-row.component.html]: All the other buttons use the tui spacing, I fixed it on this one but do we want to add a task to trello to change them over to tailwind?

**Review by keyvankhademi — APPROVED** [review]

**Pull request merged**